bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -513,6 +514,9 @@ public static boolean isReferenceEqual(Object lhsValue, Object rhsValue) {
                 return lhsType.getPackage().equals(rhsType.getPackage()) &&
                         lhsType.getName().equals(rhsType.getName()) && rhsType.equals(lhsType);
             default:
+                if (lhsValue instanceof RegExpValue && rhsValue instanceof RegExpValue) {
```

Comment:
Can't we use the typeTags?

Reply:
Since we don't have a separate runtime type to represent the regexp we can't really use a type tag? The runtime type of the regexp value is taken as `anydata & readonly`